examples/fizzbuzz/main.c

```diff
@@ -0,0 +1,189 @@
+    free(fb);
+}
+
+
+qmlbind_value *fizzbuzz_call_method_cb(qmlbind_engine *engine,
+                                       qmlbind_client_object *obj,
+                                       const char *name,
+                                       int argc,
+                                       const qmlbind_value *const *argv)
+{
+    fizzbuzz_t *fb = (fizzbuzz_t *) obj;
+
+    if (strcmp(name, "onInputChanged") == 0 && argc == 1) {
+        on_changed(fb, argv[0]);
+        return qmlbind_value_new_undefined();
+    } else if (strcmp(name, "quit") == 0) {
+        exit(0); // FIXME: should call app.quit()
+    } else {
+        return qmlbind_value_new_undefined();
+    }
+}
+
+
+qmlbind_value *fizzbuzz_get_property_cb(qmlbind_engine *engine,
+                                        qmlbind_client_object *obj,
+                                        const char *name)
+{
+    fizzbuzz_t *fb = (fizzbuzz_t *) obj;
+
+    if (strcmp(name, "input")) {
+        return qmlbind_value_new_number(fb->input);
+    } else if (strcmp(name, "result")) {
+        return qmlbind_value_new_string(strlen(fb->result), fb->result);
+    } else {
+        return qmlbind_value_new_null();
+    }
+}
+
+
+void fizzbuzz_set_property_cb(qmlbind_engine *engine,
+                              qmlbind_client_object *obj,
+                              const char *name,
+                              const qmlbind_value *value)
+{
+    fizzbuzz_t *fb = (fizzbuzz_t *) obj;
+
+    if (strcmp(name, "input")) {
+        fb->input = (int) qmlbind_value_get_number(value);
+
+        qmlbind_value params[1] = { qmlbind_value_new_number(fb->input) };
+        qmlbind_signal_emitter_emit(fb->signalEmitter, "inputChanged", 1, params);
+    } else if (strcmp(name, "result")) {
+        qmlbind_string *qstr = qmlbind_value_get_string(value);
+        const char *str = qmlbind_string_get_chars(qstr);
+        strcpy(fb->result, str);
+
+        qmlbind_value params[1] = { qmlbind_value_new_string(strlen(fb->result), fb->result) };
+        qmlbind_signal_emitter_emit(fb->signalEmitter, "resultChanged", 1, params);
+    }
+}
+
+
+void create_fizzbuzz_metaclass()
+{
+    qmlbind_client_class *classRef = NULL;
+    qmlbind_client_callbacks callbacks;
+    callbacks.new_object = fizzbuzz_new_object_cb;
+    callbacks.delete_object = fizzbuzz_delete_object_cb;
+    callbacks.call_method = fizzbuzz_call_method_cb;
+    callbacks.get_property = fizzbuzz_get_property_cb;
+    callbacks.set_property = fizzbuzz_set_property_cb;
+
+    qmlbind_metaclass *metaclass =
+        qmlbind_metaclass_new(classRef, "FizzBuzz", callbacks);
+
+    char *params1[] = { "input" };
+    qmlbind_metaclass_add_signal(metaclass, "inputChanged", 1, params1);
+    qmlbind_metaclass_add_property(metaclass, "input", "input");
+
+    char *params2[] = { "result" };
+    qmlbind_metaclass_add_signal(metaclass, "resultChanged", 1, params2);
+    qmlbind_metaclass_add_property(metaclass, "result", "result");
```

Comment:
Again, second string argument should be `resultChanged` to match the signal name.